driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -242,6 +245,7 @@ private BsonDocument getCommandToEncode() {
         }
         boolean firstMessageInTransaction = sessionContext.notifyMessageSent();
         if (sessionContext.hasActiveTransaction()) {
+            checkServerVersionForTransactionSupport();
```

Comment:
Add a unit test in com.mongodb.internal.connection.CommandMessageSpecification.  Probably similar to 'should throw if payload document bigger than max document size'.